rs/state_manager/src/checkpoint.rs

```diff
@@ -137,6 +137,8 @@ pub(crate) fn make_checkpoint(
         )?
     };
 
+    cp.remove_unverified_checkpoint_marker()?;
+
```

Comment:
That's a good question.
The debug_assertion in manifest computation always makes tests fail. If we remove this one, there are still many tests failing. 

Some are due to not being able to get StateSyncMessage because unverified checkpoint cannot be seen. 

Also, many assertions about checkpoint_heights fails because checkpoint_heights does not include unverified checkpoints.

Reply:
Since we agree on crashing during manifest computation, these tests will fail at an earlier state.

Reply:
Then it's probably worth it to add a check into checkpoint unit tests somewhere